### DISCUSSION TITLE
How to await for vercel deployment in github actions

### DISCUSSION THREAD

**Discussion by virendra038** [community/community#51562] [Actions] [answered]
### Select Topic Area

Question

### Body

Need to run cypress tests after vercel does the deployment ?
github repo is linked to my vercel account so vercel is auto deploying everytime i push my code to github
Any idea how to achieve that ?

**Comment by Anmol-Baranwal** [comment, marked as answer]
To run Cypress tests after Vercel deploys your code, you can use a Continuous Integration (CI) tool like GitHub Actions or Travis CI. These tools can be configured to automatically run tests after a deployment is completed, allowing you to verify that your code is working as expected.

Here's a general approach you can follow to set this up:

1. First, you'll need to set up a CI workflow in your GitHub repository that will run your Cypress tests after a deployment is completed. You can use a tool like GitHub Actions to do this. GitHub Actions provides a wide range of pre-built workflows that you can use to get started quickly.

2. Configure the workflow to trigger after a Vercel deployment: In your workflow file, you'll need to configure it to trigger after a Vercel deployment is completed. You can do this by adding a trigger to the workflow file that will listen for a Vercel deployment event.

3. Install Cypress in your CI environment: You'll need to make sure that Cypress is installed in your CI environment so that your tests can run. You can do this by adding a step to your workflow that will install Cypress.

4. Run your Cypress tests: Finally, you'll need to add a step to your workflow that will run your Cypress tests. You can do this by running the Cypress CLI command in your workflow.

Once you have this set up, your tests will automatically run after each deployment, allowing you to catch any issues that may have been introduced during the deployment process.

**Reply by Anmol-Baranwal** [reply]
@virendra038 
I wanted to follow up and ask if you have any additional questions or concerns about my previous response on GitHub. If you have no further inquiries, would you kindly consider marking my answer as accepted?

It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

**Reply by cd3vane** [reply]
I have everything else set up except step 2. I cannot get vercel deployment event to trigger the pipeline, I have tried adding an webhook on vercel to call the pipeline API on gitlab but this is not working

**Reply by psahni** [reply]
> * Configure the workflow to trigger after a Vercel deployment: In your workflow file, you'll need to configure it to trigger after a Vercel deployment is completed. You can do this by adding a trigger to the workflow file that will listen for a Vercel deployment event.
> *

Configure the workflow to trigger after a Vercel deployment: In your workflow file, you'll need to configure it to trigger after a Vercel deployment is completed. You can do this by adding a trigger to the workflow file that will listen for a Vercel deployment event.

@Anmol-Baranwal 
what is that configuration. request you to talk in terms of code also

**Comment by triozer** [comment]
To run end-to-end tests after deployment, you can follow this guide provided by Vercel.

https://vercel.com/guides/how-can-i-run-end-to-end-tests-after-my-vercel-preview-deployment
